Controllers/Admin/TaskController.php

```diff
@@ -0,0 +1,157 @@
+<?php
+declare(strict_types=1);
+
+namespace Controllers\Admin;
+
+use Controllers\BaseController;
+use Models\TaskModel;
+use View\TaskView;
+
+/**
+ * @package Controllers\Admin
+ */
+class TaskController extends BaseController
+{
+    /**
+     * @var TaskModel
+     */
+    private $taskModel;
+
+    /**
+     * @var TaskView
+     */
+    private $taskView;
+
+    public function __construct()
+    {
+        $this->taskModel = new TaskModel();
+        $this->taskView = new TaskView();
+    }
+
+    /** Render admin panel by param
+     * @param array $param
+     */
+    public function formAction(array $param): void
+    {
+        $access = $this->taskModel->isAccess();
+        if ($access) {
+            switch ($param[0]) {
+                case 'delete' :
+                    $data = $this->taskModel->adminGetAllTasks($access);
+                    $options = ['title' => 'Удаление',
+                        'content' => 'admin/delete_task.phtml',
+                        'data' => $data];
+                    break;
+                case 'change' :
+                    $data = $this->taskModel->adminGetAllTasks($access);
+                    $options = ['title' => 'Изменить',
+                        'content' => 'admin/change_task.phtml',
+                        'data' => $data];
+                    break;
+                case 'approve' :
+                    $data = $this->taskModel->getDoneTasks($access);
+                    $options = ['title' => 'Подтвердить',
+                        'content' => 'admin/approve_task.phtml',
+                        'data' => $data];
+                    break;
+                case 'restart' :
+                    $data = $this->taskModel->getFailTasks($access);
+                    $options = ['title' => 'Возобновить',
+                        'content' => 'admin/fail_task.phtml',
+                        'data' => $data];
+                    break;
+                case 'new' :
+                    $data = $this->taskModel->selectUsersForNewTask($access);
+                    $options = ['title' => 'Новая задача',
+                        'content' => 'admin/new_task.phtml',
+                        'data' => $data];
+                    break;
+                default :
+                    $this->homeLocation();
+            }
+            $this->taskView->render($options);
+        } else {
+            $this->homeLocation();
+        }
+    }
+
+    /**
+     * Delete task
+     * @return void
+     */
+    public function deleteAction(): void
+    {
+        if ($this->checkPost()) {
+            $result = $this->taskModel->deleteTask($_POST);
```

Comment:
please incapsulate $_POST array to a separate model that will be responsible for retrieving request parameters or headers and use it in other classes.